ocaml/forkexecd/lib/fe_stubs.c

```diff
@@ -0,0 +1,405 @@
+/*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <stdbool.h>
+#include <math.h>
+#include <pthread.h>
+#include <limits.h>
+#include <sys/wait.h>
+
+#include <caml/fail.h>
+#include <caml/memory.h>
+#include <caml/unixsupport.h>
+#include <caml/signals.h>
+
+#include "../helper/vfork_helper.h"
+
+#define FOREACH_LIST(name, list) \
+    for(value name = (list); name != Val_emptylist; name = Field(name, 1))
+
+// Create thread reducing stack usage to a minimum to reduce memory usage.
+// Returns error number (like pthread_create).
+static int create_thread_minstack(pthread_t *th, void *(*proc)(void *), void *arg);
+
+static inline void
+reap_pid(pid_t pid)
+{
+    int status;
+    while (waitpid(pid, &status, 0) < 0 && errno == EINTR)
```

Comment:
instead of reaping one pid, why not reap them all? Then we can also do this from OCaml (see my other comment).
We shouldn't have anything else other than forkexecd/forkhelpers spawning processes anyway.

Reply:
No, you would introduce races you don't want.